src/content/en/2021/caching.md

```diff
@@ -17,4 +20,425 @@ featured_stat_3: TODO
+
+{{ figure_markup(
+  image="top-cdns.png",
+  caption="Adoption of the top CDNs.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=2035524639&format=interactive",
+  height=501,
+  sheets_gid="58739923",
+  sql_file="top_cdns.sql"
+) }}
+
+While Cloudflare is used twice as much as Google, a large variety of solutions remain available, including Fastly, Amazon CloudFront, Akamai, and many others.
+
+## Service Worker adoption
+
+A Service Worker is a script that the browser allows to run in the background independently from a web page. It supports features that don't need direct user or web page interaction and offer rich offline experiences.
+
+The adoption of Service Workers has continued to steadily increase. While 1% of Mobile pages registered a service worker in 2020, 9% of Mobile pages ranked in the top 1,000 registered one in 2021.
+
+{{ figure_markup(
+  image="sw-adoption.png",
+  caption="Service Worker adoption.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=688062455&format=interactive",
+  sheets_gid="802609299",
+  sql_file="service_worker_rank.sql"
+) }}
+
+The primary way to cache resources within a Service Worker is by using the *CacheStorage API*. This allows a developer to create a custom cache strategy for any requests passing through the worker; some well-known ones are Stale-While-Revalidate, Cache Falling Back to Network, Network Falling Back to Cache, and Cache Only. In recent years it has become even easier to adopt those strategies through the increased popularity of [Workbox](https://developers.google.com/web/tools/workbox/modules/workbox-strategies), which helps you decide what cache you want to plug and play.
+
+## Headers adoption
+
+With both a CDN and the Browser, HTTP Headers are the primary toolkit a developer needs to master to properly cache resources. Headers are simply instructions known as "*directives*", read during the HTTP Request or Response to control the cache strategy.
+
+The caching-related headers, or the absence of them, tell the browser or CDN three essential pieces of information:
+- **Cacheability**: Is this content cacheable?
+- **Freshness**: If it is cacheable, how long can it be cached for?
+- **Validation**: If it is cacheable, how do I ensure that my cached version is still fresh?
+
+Headers are meant to be used either alone or together. To tell if the content is cacheable and fresh, we have:
```

Comment:
Edit: 

Headers are meant to be used either alone or together. To determine if the content is cacheable and fresh, we have:
 - `Expires` specifies an explicit expiration date and time (i.e., when precisely the content expires)
- `Cache-Control` specifies a cache duration (i.e., how long the content can be cached in the browser relative to when it was requested)
When both are specified, `Cache-Control` takes precedence.

Usage of the Cache-Control header has increased steadily since 2019. 74% of responses on Mobile pages included the Cache-Control header, while 75% of responses on Desktop pages utilized the header. From 2020, the usage of this specific header increased by 0.71% for Mobile and by 1.13% for Desktop. This leads us to believe there has been an increase in awareness in the community around proper usage of Cache-Control, notably for Desktop.